src/EPiServer.SocialAlloy.Web/Social/Tutorials/LikeButton/LikeButtonBlockController.cs

```diff
@@ -0,0 +1,150 @@
+
+            // Create a rating block view model to fill the frontend block view
+            var blockModel = new LikeButtonBlockViewModel();
+
+            try
+            {
+                // Using the EPiServer Social Rating service, get the existing rating for the current user (rater) and page (target).
+                // Again, if there's no user logged into the site then we generate a random anonymous user, which should not currently 
+                // have any ratings, thus he/she will always be allow to rate the current page.
+                var ratingPage = ratingService.Get(
+                    new Criteria<RatingFilter>()
+                    {
+                        Filter = new RatingFilter()
+                        {
+                            Rater = Reference.Create(raterUser),
+                            Targets = new List<Reference> { Reference.Create(targetPage) }
+                        },
+                        PageInfo = new PageInfo() { PageSize = 1 }
+                    }
+                );
+
+                // Add the current Like rating, if any, to the block view model. If the user is logged 
+                // into the site and had previously liked the current page then the CurrentRating value
+                // should be 1 (LIKED_RATING).  Anonymous user Likes are generated with unique random users
+                // and thus the current anonymous user will never see a current rating value as he/she
+                // can Like the page indefinitely.
+                if (ratingPage.Results.Count() > 0)
+                {
+                    blockModel.CurrentRating = ratingPage.Results.ToList().FirstOrDefault().Value.Value;
+                }
+
+                // Using the EPiServer Social Rating service, get the existing Like statistics for the page (target)
+                var ratingStatisticsPage = ratingService.Get(
+                    new Criteria<RatingStatisticsFilter>()
+                    {
+                        Filter = new RatingStatisticsFilter()
+                        {
+                            Targets = new List<Reference> { Reference.Create(targetPage) }
+                        },
+                        PageInfo = new PageInfo() { PageSize = 1 }
+                    }
+                );
+
+                // Add the page Like statistics to the block view model
+                if (ratingStatisticsPage.Results.Count() > 0)
+                {
+                    var statistics = ratingStatisticsPage.Results.ToList().FirstOrDefault();
+                    if (statistics.TotalCount > 0)
+                    {
+                        blockModel.TotalCount = statistics.TotalCount;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // the rating service may throw a number of possible exceptions
+                // should handle each one accordingly -- see rating service documentation
```

Comment:
I rather just keep any external links in README...agree?  Doesn't the main README already have a link?  If not, we should probably add it?